tests/checkasm/vvc_deblock.c

```diff
@@ -0,0 +1,160 @@
+
+static const uint32_t pixel_mask[3] = {0xffffffff, 0x03ff03ff, 0x0fff0fff};
+
+#define SIZEOF_PIXEL ((bit_depth + 7) / 8)
+#define BUF_SIZE 16 * 16
+
+#define randomize_buffers(buf0, buf1, size)               \
+    do                                                    \
+    {                                                     \
+        uint32_t mask = pixel_mask[(bit_depth - 8) >> 1]; \
+        int k;                                            \
+        for (k = 0; k < size; k += 4 / sizeof(*buf0))     \
+        {                                                 \
+            uint32_t r = rnd() & mask;                    \
+            AV_WN32A(buf0 + k, r);                        \
+            AV_WN32A(buf1 + k, r);                        \
+        }                                                 \
+    } while (0)
+
+#define TC25(x) ((tc[x] * 5 + 1) >> 1)
+#define MASK(x) (uint16_t)(x & ((1 << (bit_depth)) - 1))
+#define GET(x) ((SIZEOF_PIXEL == 1) ? *(uint8_t *)(&x) : *(uint16_t *)(&x))
+#define SET(x, y)                  \
+    do                             \
+    {                              \
+        uint16_t z = MASK(y);      \
+        if (SIZEOF_PIXEL == 1)     \
+            *(uint8_t *)(&x) = z;  \
+        else                       \
+            *(uint16_t *)(&x) = z; \
+    } while (0)
+#define RANDCLIP(x, diff) av_clip(GET(x) - (diff), 0,       \
+                                  (1 << (bit_depth)) - 1) + \
+                              rnd() % FFMAX(2 * (diff), 1)
+
+#define P3 buf[-4 * xstride]
+#define P2 buf[-3 * xstride]
+#define P1 buf[-2 * xstride]
+#define P0 buf[-1 * xstride]
+#define Q0 buf[0 * xstride]
+#define Q1 buf[1 * xstride]
+#define Q2 buf[2 * xstride]
+#define Q3 buf[3 * xstride]
+
+static void check_deblock_chroma_horizontal()
+{   
+    int32_t tc[4] = {(rnd() & 393) + 3, 600, (rnd() & 393) + 3, (rnd() & 393) + 3};
+
+    uint8_t no_p[4] = {0, 0, 0, 0};
+    uint8_t no_q[4] = {0, 0, 0, 0};
+    uint8_t max_len_p[4] = {1, 1, 1, 1};
+    uint8_t max_len_q[4] = {1, 3, 3, 1};
+    int shift = 0;
+    int beta[4] = {(rnd() & 81) + 8, (rnd() & 81) + 8, (rnd() & 81) + 8, (rnd() & 81) + 8 };
+
+    LOCAL_ALIGNED_32(uint8_t, buf0, [BUF_SIZE]);
+    LOCAL_ALIGNED_32(uint8_t, buf1, [BUF_SIZE]);
+
+    VVCDSPContext context;
+    int bit_depth;
+
+    declare_func(void, uint8_t *pix, ptrdiff_t stride, const int32_t *beta, const int32_t *tc,
+                 const uint8_t *no_p, const uint8_t *no_q, const uint8_t *max_len_p, const uint8_t *max_len_q, int shift);
+
+    const int size = shift ? 4 : 2;
+    const int end = 8 / size;
+
+    for (bit_depth = 8; bit_depth <= 12; bit_depth += 2)
+    {
+        int xstride = SIZEOF_PIXEL * 8; // bytes
+        int ystride = 2;                // bytes
+
+        ff_vvc_dsp_init(&context, bit_depth);
+        if (check_func(context.lf.filter_chroma[0], "vvc_h_loop_filter_chroma_%d", bit_depth))
+        {
+            int i, j, b3, tc25, tc25diff, b3diff;
```

Comment:
better move this to randomize_chroma_buffers. and follow the structure in https://github.com/ffvvc/FFmpeg/blob/e81b6d78fc2ddf8edd53a6a052713354ef8d27c2/tests/checkasm/hevc_deblock.c#L48